.github/scripts/collect_github_metrics.py

```diff
@@ -102,18 +112,38 @@ def main():
     repo = g.get_repo(repo_name)
 
     run = repo.get_workflow_run(int(run_id))
-
-    workflow_data_query = f'''INSERT INTO github_workflow_runs_test(
+    if run.status != 'completed':
+        logger.error('Run %s is not completed! Only completed runs should be in the database', run_id)
+        raise SystemExit(1)
+
+    # We rely on the following assumptions:
+    # - The workflow run is completed. When run.status != 'completed' we should not add it to the database
+    #   theoretically the second attempt can be triggerred right after the completion of the first one
+    #   or while the runner which executes this script is deploying
+    #
+    # - Total duration of a workflow run equals "run.updated_at - run.run_started_at"
```

Comment:
It seems to be true for the pipelines with one attempt and no retries. However, if there are 2 or more attempts, the `updated_at` time is, firstly, set to the time the pipeline is re-triggered and then, after the pipeline completion, it is set to the completion time for the *latest* attempts as they share the same `workflow_run_id`.

e.g., for https://github.com/openvinotoolkit/openvino/actions/runs/8100288997:
* The first attempt started on `2024-02-29T17:40:21Z` and was completed the same day
* I retriggered it and the new `updated_at` is `2024-03-21T13:42:43Z` - the time the second attempt started
* Once it finishes, the `updated_at` time will be different again
* If we calculate `run.updated_at - run.run_started_at` after the second attempt finishes, the total pipeline time will be several days, almost a month

Reply:
I don't see the problem you're talking about in the run you've provided (https://api.github.com/repos/openvinotoolkit/openvino/actions/runs/8100288997)
Once the run is completed (and we don't want to run the script otherwise), whatever the attempt number is, `run_started_at` shows the start time of the job restarted and `updated_at` corresponds with the time the job is finished.

Reply:
Okay, there's actually an API endpoint for that 😅 https://api.github.com/repos/openvinotoolkit/openvino/actions/runs/8100288997/timing